docs/sophia/containers/containers.md

```diff
@@ -0,0 +1,55 @@
+module use /soft/spack/base/0.7.1/install/modulefiles/Core/module load apptainer
+apptainer version #1.3.3
+```
+
+The Apptainer version on Sophia is 1.3.3. Detailed user documentation is available [here](https://apptainer.org/docs/user/1.3/)
+
+## Building from Docker or Argonne GitHub Container Registry
+
+Containers on Sophia can be built by writing Dockerfiles on a local machine and then publish the container to DockerHub, or by directly building them on ALCF compute node by writing an Apptainer recipe file. If you prefer to use existing containers, you can pull them from various registries like DockerHub and run them on Sophia.
+
+Since Docker requires root privileges, which users do not have on Sophia, existing Docker containers must be converted to Apptainer. To build a Docker-based container on Sophia, use the following as an example:
+
+
+```bash
+qsub -I -A <Project> -l select=1:ngpus=8:ncpus=256 -l walltime=01:00:00 -l filesystems=home:eagle -l singularity_fakeroot=True -q by-node -k doe
+export HTTP_PROXY=http://proxy.alcf.anl.gov:3128
+export HTTPS_PROXY=http://proxy.alcf.anl.gov:3128
+export http_proxy=http://proxy.alcf.anl.gov:3128
+export https_proxy=http://proxy.alcf.anl.gov:3128
+module use /soft/spack/base/0.7.1/install/modulefiles/Core/module load apptainer
+apptainer build --fakeroot pytorch:22.06-py3.sing docker://nvcr.io/nvidia/pytorch:22.06-py3
+```
+You can find the latest prebuilt Nvidia PyTorch containers [here](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch).  The Tensorflow containers are [here](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tensorflow) (though note that LCF doesn't prebuild the TF-1 containers typically).  You can search the full container registry [here](https://catalog.ngc.nvidia.com/containers). For custom containers tailored for Sophia, visit [ALCF's GitHub container registry](https://github.com/argonne-lcf/container-registry/tree/main)
+
+> **Note:** Currently container build and executions are only supported on the Sophia compute nodes
+
+## Recipe-Based Container Building
+
+As mentioned earlier, you can build Apptainer containers from recipe files. Instructions are available [here](https://apptainer.org/docs/user/1.3/build_a_container.html#building-containers-from-apptainer-definition-files).
+
+> Note: You can also build custom recipes by bootstrapping from prebuilt images. For e.g the first two lines in a recipe to use our custom Tensorflow implementation would be `Bootstrap: oras` followed by `From: ghcr.io/argonne-lcf/tf2-mpich-nvidia-gpu:latest`
+
+## Troubleshooting Common Issues
+
+- **Permission Denied Error**: If you encounter permission errors during the build
+	- Check your quota and delete any unnecessary files. 
+	- Clean-up apptainer cache, `~/.apptainer/cache`, and set the apptainer tmp and cache directories as below. If your home directory is full and if you are building your container on a compute node, then set the tmpdir and cachedir to local scratch 
```

Comment:
same comment as in `polaris/containers.md`

Not exclusive to this PR, but a more general comment: we should make an effort to cut down on some of the repeated machine-independent documentation, e.g. issues that affect Apptainer containers on all machines, the generic part of "Recipe-Based Container Building" section, etc. 

This can be accomplished using [`pymdownx.snippets`](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/), already demonstrated in this repo in #177.

One option is to have all of the generic content in some main `containers.md`, either excluded from navigation or placed in a new page Services > Containers. Then both `polaris/containers.md` and `sophia/containers.md` can use the  `--8<--` magics to reference that main file. Another option is to keep this information in `polaris/containers.md` and have `sophia/containers.md` pull from it.

Reply:
maintaining the repeated content will be more annoying once Aurora, Crux, etc. docs are also in production-mode

Reply:
I think this is a good idea. I will implement it